### DISCUSSION TITLE
I use Chinese Q&A, but the answers are in English. How should I solve this problem?

### DISCUSSION THREAD

**Discussion by Mioooooo** [vanna-ai/vanna#423] [Q&A] [answered]
ollama ---> qwen:32b
qwen is a large model open sourced by the Chinese company Alibaba. When I use the ollama console to ask qwen questions in Chinese, he always replies in Chinese, but this is not the case in vanna. What should I do? Prompt?

**Comment by zainhoda** [comment, marked as answer]
In `v0.5` you can now set a "language" parameter.

`config={..., 'language': 'Mandarin'}`

Could you try that and see if it works?

**Reply by Mioooooo** [reply]
'language': 'Chinese'？

**Reply by lemon3853** [reply]
> In `v0.5` you can now set a "language" parameter.
> 
> `config={..., 'language': 'Mandarin'}`
> 
> Could you try that and see if it works?

I configured 'language': 'Chinese' or 'language': 'China', but they are not taking effect.

**Comment by lemon3853** [comment]
> ollama ---> qwen:32b qwen is a large model open sourced by the Chinese company Alibaba. When I use the ollama console to ask qwen questions in Chinese, he always replies in Chinese, but this is not the case in vanna. What should I do? Prompt?



> ollama ---> qwen:32b qwen is a large model open sourced by the Chinese company Alibaba. When I use the ollama console to ask qwen questions in Chinese, he always replies in Chinese, but this is not the case in vanna. What should I do? Prompt?

对大模型的提问在代码里用英文写死了，只能修改源码，用中文提问。参考：base.py generate_summary方法

**Reply by Mioooooo** [reply]
我目前是这样解决的，但是每次更新后都要重新写一边，有点麻烦

**Reply by niklenikkle** [reply]
你好 可以看一下你的代码怎么写的嘛

**Reply by xuxin-xxu** [reply]
> 你好 可以看一下你的代码怎么写的嘛

直接改flask里面的assets.py

**Comment by WZAZSY** [comment]
def tenant_safe_prompt(tenant_id: str) -> str:
    return f"""
    你是MySQL专家。请帮助生成一个SQL查询来回答这个问题。你的回答应该只基于给定的上下文，并遵循回答指南和格式说明。重要约束如下：
    1. 所有查询必须包含 cust_no = '{tenant_id}'
    2. 生成数据结果的字段名需转为中文：
    """


# 在 MyVanna 配置中使用
vn = MyVanna(
    {
        "api_key": "sk-da30fa08991c4de3ab198a42c8c0665e",
        "model": "deepseek-chat",
        "path": "/chromadb",
        "language": "zh",
        "dialect": "mysql",
        "initial_prompt": tenant_safe_prompt( "K00002"),
    }
)

vn.add_documentation("DRIVER字段在输出的sql中，别名为司机姓名 ")


输出的sql：
 SELECT
    CO_PICK_NO AS '提货的提单单据号',
    CO_ORDER_NO AS '订单号',
    MONTH_CONTRACT_ID AS '月底订单号码',
    YEAR_CONTRACT_ID AS '年度订单号码',
    CUST_NAME AS '客户的名称',
    DEPO_NAME AS '货物出库仓库名称',
    SCCJ_NAME AS '货物的产地',
    TYPE_NAME AS '货物品种名称',
    STEEL_NUM AS '货物牌号',
    SPECS AS '货物规格',
    CO_PICK_QTY AS '提货的重量',
    CO_PICK_PIE AS '提货的件数',
    CO_CK_QTY AS '提单的出库重量',
    CO_CK_PIE AS '提单的出库件数',
    TRANS_DATE AS '提单的提货日期或开单日期',
    INV_CK_DATE AS '提单的出库日期',
    DRIVER AS '司机姓名',
    DRIVER_ID AS '运输司机的身份证号',
    CAR_SHIP_NO AS '运输车辆的车牌号',
    DRIVER_PHONE AS '运输司机的手机号码',
    NOTES AS '提单的备注',
    SH_NAME AS '收货单位名称',
    SH_LXR AS '收货人联系人姓名',
    SH_LXR_TEL AS '收货联系人的电话号码',
    CUST_BILL_NO AS '客户的单据号'
FROM
    CX034
WHERE
    CUST_NO = 'K00002'
    AND YEAR(TRANS_DATE) = 2025
LIMIT 10;
